source/development/feature-flags.html.md

```diff
@@ -0,0 +1,76 @@
+
+Feature flags are more useful for wide scope changes that are likely to take a long time to action over difficult to implement changes. For difficult to implement but short term changes, consider using a pre-release instead (see below section on pre-releases).
+
+### We are unsure of the thing’s value
+This is an alternative to the previous principal on implementation difficulty where, for whatever reason, we have a significant change that we aren’t 100% confident in and have been unable to verify its value. In this instance, putting it behind a feature flag would allow us to get early feedback from users who’re willing to use and test our new/updated feature and give us feedback, allowing us to iterate the change with data.
+
+## When a feature flag isn't useful
+Even if you have a scenario where feature flags may be useful, consider the following before implementing one.
+
+### When releasing the thing carries no significant risk
+We should avoid using feature flags in most cases. We have alternative systems for communicating big or impactful changes such as feature depreciation where we tell people we’ll be removing something in a future breaking release and breaking releases themselves where users know that by upgrading, they might need to change something in their service. Feature flags should be used as a way to alleviate instances where releasing as normal is challenging.
+
+### We could use a pre-release instead
+Another method we have for testing breaking changes before release are pre-releases. This is a version of govuk-frontend that we release to npm like normal releases, but we tag it as a “beta” release eg: instead of 5.0.0, we might release 5.0.0-beta.0. This communicates that it isn’t the next major version of govuk-frontend but an early, “experimental” version of that release, allowing users to assess it before we release that major version.
+
+Pre-releases are ideal for short term investigation or early access to the community when a breaking change might include something difficult to implement. They’re also generally easier to manage compared to feature flags. Consider if you could use a pre-release instead of a feature flag.
+
+## Types of feature flag
+There are 3 broad types of feature flag. These are defined more by the change you’re making rather than implementation of the flag itself.
+
+### Experimental
+Early previews of work that will likely go into a future release. Enables us to receive feedback on something we’re not 100% confident in from members of the community who’re willing to test it in their service. Use an experimental flag to release a long term change early to assess its value in a live environment.
```

Comment:
```suggestion
Use a flag to create early previews of a future release to help assess its value in a live environment. In this case the flag will enable you to receive feedback from members of the community who’re willing to test early versions in their service.
```

Reply:
Some of the content in this doc reads as duplication. Is it worth looking at ways to merge any duplication or is it repeated for contextual reasons?

Reply:
I'm guessing you're referring to the stuff in this section where I define the different classifications of feature flag, which I then go into again when defining when to use a feature flag at all. Probably we can merge them. Knowing the types of feature flags is potentially useful if we end up using them more so that we can establish common language for our users, however it's not really necessary.

The secret main reason this section exists is to house the "extended deprecation" heading and tell people not to delay deleting a feature unless we absolutely have to. Probably we can convey this somewhere else.

Reply:
Honestly, I've forgotten what I was specifically referring to. If re-emphasising something is important for the context, then let's not shuffle content around unnecessarily. We can always do some lite testing of how people read/use the content to double-check it's clear enough.

Reply:
Lets leave it for now and see how we get on